platform/npm/tests/request-data.js

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+
+    uBlock Origin - a browser extension to block requests.
+    Copyright (C) 2014-present Raymond Hill
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see {http://www.gnu.org/licenses/}.
+
+    Home: https://github.com/gorhill/uBlock
+*/
+
+'use strict';
+
+/******************************************************************************/
+
+import { strict as assert } from 'assert';
+import fs from 'fs/promises';
+import { createRequire } from 'module';
+
+const require = createRequire(import.meta.url);
+
+const requests = require('scaling-palm-tree/requests.json');
+const results = require('../results.json');
+
+const typeMap = {
+    document: 'sub_frame',
+    stylesheet: 'stylesheet',
+    image: 'image',
+    media: 'media',
+    font: 'font',
+    script: 'script',
+    xhr: 'xmlhttprequest',
+    fetch: 'xmlhttprequest',
+    websocket: 'websocket',
+    ping: 'ping',
+
+    other: 'other',
+    eventsource: 'other',
+    manifest: 'other',
+    texttrack: 'other',
+};
+
+let engine = null;
+
+describe('Request data', () => {
+    before(async () => {
+        const globals = { URL, setTimeout, clearTimeout };
+
+        const { StaticNetFilteringEngine } = await import('../index.js');
+
+        engine = await StaticNetFilteringEngine.create();
+
+        await engine.useLists([
+            { name: 'easylist', raw: require('../data/easylist.json') },
+            { name: 'easyprivacy', raw: require('../data/easyprivacy.json') },
+        ]);
+    });
+
+    for ( let i = 0; i < requests.length; i++ ) {
+        const { url, frameUrl, cpt } = requests[i];
+        const request = { url, originURL: frameUrl, type: typeMap[cpt] };
+
+        const expected = results[i];
+
+        it(`should ${expected === 1 ? 'block' : 'allow'} ${request.type} URL ${request.url} from origin ${request.originURL}`, () => {
```

Comment:
If a test fails it should say something like "should block script URL https://example.com/foo.js from origin https://example.net/index.html" followed by the stack trace.